### PR TITLE
rewrite and comment determine_local_basis

### PR DESCRIPTION
**Pull request by JulienLefevreMars**
Hello,

I have rewritten and commented determine_local_basis in curvature.py . I also removed the proj_matrix inside curvature_fit since it was used only to compute the local basis.

### CODE REVIEW

**Review by MaximeDdnn — COMMENTED** [review]
- i like the idea to move the proj_matrix declaration from the curvature.py to determine_local_basis.py
- need to add precision of the shape of vec1 vec2 in the doc string

**Comment by JulienLefevreMars** [comment]
Hey @MaximeDdnn ,

I added precisions on the code and unit tests. Hope it will be clear enough :-)

**Review by alexpron — CHANGES_REQUESTED** [review]
Hello @JulienLefevreMars thank you for contribution that clearly improve the quality of the code.
Just to be more formal the docstring should follow the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) format.

File: slam/curvature.py [hunk 1]
```diff
@@ -87,18 +87,18 @@ def curvature_fit(mesh, tol=1e-12, neighbour_size=2):
 
 def determine_local_basis(normal, tol):
     """
-    Computes the local basis of the tangent plane
-    :param normal: (3,1) normal vector
+    Computes the orthonormal local basis (vec1, vec2) of the tangent plane
+    :param normal: (3,1) normal vector, should be unitary
     :param tol: tolerance value
     :return:
```
Comment (gauzias) [inline slam/curvature.py:93]: please complete the docstring with the description of the ouput vectors

**Review by gauzias — CHANGES_REQUESTED** [review]
thanks @JulienLefevreMars were can almost merge, but please complete the docstring.

File: slam/curvature.py (hunk 1 above)
Comment (JulienLefevreMars) [reply slam/curvature.py:93]: I added some comments for the return. It is a good thing to make the code the more explicit as possible, but "explicit" has also to be explicit (please add standards/rules somewhere in order to know exactly what to do in a PR, in commenting/docstringing the code etc).

Comment (alexpron) [reply slam/curvature.py:93]: Nice ty @JulienLefevreMars I am currently trying to add modifications to your PR but i am still struggling for the push. Hope i l solve it soon

Comment (JulienLefevreMars) [reply slam/curvature.py]: it's great @alexpron , it provides a good canvas for how to write a proper docstring, thks !

**Pull request merged**